raiden/network/proxies/secret_registry.py

```diff
@@ -88,61 +120,175 @@ def register_secret_batch(
             return
 
         checking_block = self.client.get_checking_block()
-        error_prefix = 'Call to registerSecretBatch will fail'
         gas_limit = self.proxy.estimate_gas(checking_block, 'registerSecretBatch', secrets)
+        receipt = None
+        transaction_hash = None
+        msg = None
+
         if gas_limit:
-            error_prefix = 'Call to registerSecretBatch failed'
+            gas_limit = safe_gas_limit(
+                gas_limit,
+                len(secrets) * GAS_REQUIRED_PER_SECRET_IN_BATCH,
+            )
+
+            log.debug('registerSecretBatch called', **log_details)
+
             try:
-                gas_limit = safe_gas_limit(
+                transaction_hash = self.proxy.transact(
+                    'registerSecretBatch',
                     gas_limit,
-                    len(secrets) * GAS_REQUIRED_PER_SECRET_IN_BATCH,
+                    secrets_to_register,
                 )
-                transaction_hash = self.proxy.transact('registerSecretBatch', gas_limit, secrets)
                 self.client.poll(transaction_hash)
-                receipt_or_none = check_transaction_threw(self.client, transaction_hash)
-            except Exception as e:
-                secret_registry_transaction.set_exception(e)
-                msg = 'Unexpected exception at sending registerSecretBatch transaction'
+                receipt = self.client.get_transaction_receipt(transaction_hash)
+            except Exception as e:  # pylint: disable=broad-except
+                transaction_result.set_exception(e)
+                msg = f'Unexpected exception {e} at sending registerSecretBatch transaction.'
             else:
-                secret_registry_transaction.set(transaction_hash)
-            finally:
-                for secret in secrets_to_register:
-                    self.open_secret_transactions.pop(secret, None)
-
-        transaction_executed = gas_limit is not None
-        if not transaction_executed or receipt_or_none:
-            if transaction_executed:
-                block = receipt_or_none['blockNumber']
-            else:
-                block = checking_block
+                transaction_result.set(transaction_hash)
+
+        # Clear `open_secret_transactions` regardless of the transaction being
+        # successfully executed or not.
+        with self._open_secret_transactions_lock:
+            for secret in secrets_to_register:
+                self.open_secret_transactions.pop(secret)
```

Comment:
Not doing the cleanup on failure was also a bug, because if without the flag `--unrecoverable-error-should-crash` the node would not crash and the `open_secret_transactions` would be left in a dirty state.